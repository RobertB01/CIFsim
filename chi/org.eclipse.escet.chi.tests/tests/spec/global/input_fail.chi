################################################################################
# Copyright (c) 2010, 2024 Contributors to the Eclipse Foundation
#
# See the NOTICE file(s) distributed with this work for additional
# information regarding copyright ownership.
#
# This program and the accompanying materials are made available under the terms
# of the MIT License which is available at https://opensource.org/licenses/MIT
#
# SPDX-License-Identifier: MIT
#################################################################################

proc P1(file f):
    select eof(f) end
end

proc P2(file f):
    select read(f, bool) end
end

proc P3(file f):
    select eol(f) end
end

proc P4(file f):
    select newlines(f) end
end
