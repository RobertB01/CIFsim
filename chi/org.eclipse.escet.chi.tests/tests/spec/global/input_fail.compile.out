Error: File "spec/global/input_fail.chi": Semantic error at line 14, column 12: "eof" is not allowed here, the expression must be constant.
Error: File "spec/global/input_fail.chi": Semantic error at line 18, column 12: "read" is not allowed here, the expression may not have side-effects.
Error: File "spec/global/input_fail.chi": Semantic error at line 22, column 12: "eol" is not allowed here, the expression must be constant.
Error: File "spec/global/input_fail.chi": Semantic error at line 26, column 12: "newlines" is not allowed here, the expression must be constant.
